lib/sparkle_formation/sparkle_attribute.rb

```diff
@@ -300,20 +300,23 @@ def _raise(*args)
     end
     alias_method :raise!, :_raise
 
-    # @return [TrueClass, FalseClass] resource can be tagged
+    # @return [TrueClass, FalseClass, NilClass] resource can be tagged
     def taggable?
-      if(defined?(SfnAws))
-        if(self[:type])
+      if defined?(SfnAws)
+        if self[:type]
           resource = SfnAws.lookup(self[:type].gsub('::', '_').downcase)
           resource && resource[:properties].include?('Tags')
         else
-          if(self._parent)
-            self._parent.taggable?
-          end
+          _parent && _parent.taggable?
         end
       end
     end
 
+    # Tag a resource with a hash instead of repetetive key value pairs
+    def tags!(hash)
+      tags hash.map { |k, v| { 'Key' => Bogo::Utility.camel(k), 'Value' => v } }
+    end
+
```

Comment:
Helper methods follow a specific implementation pattern. Methods are defined with leading underscore: `_tags` and are then aliased to a matching bang method `tags!`. 

Automatic casing conversions are based on type provided, with symbols being auto-converted. A preferred implementation would be:

```ruby
def _tags(hash)
  tags hash.map do |k, v|
    key = k.is_a?(Symbol) ? _process_key(k, :force) : k
    {'Key' => key, 'Value' => v}
  end
end
```